realm/realm-library/src/androidTestObjectServer/java/io/realm/SchemaTests.java

```diff
@@ -145,4 +147,54 @@ public void allow_addField() {
 
         realm.close();
     }
+
+    @Test
+    public void addPrimaryKey_notAllowed() {
+        String className = "StringOnly";
+        Realm realm = Realm.getInstance(config);
+
+        realm.beginTransaction();
+        RealmObjectSchema objectSchema = realm.getSchema().get(className);
+        objectSchema.addField("foo", String.class);
+
+        try {
+            objectSchema.addPrimaryKey("foo");
+            fail();
+        } catch (UnsupportedOperationException ignored) {
+        } finally {
+            realm.commitTransaction();
+            realm.close();
+        }
+    }
+
+    @Test
+    public void addField_withPrimaryKeyModifier_notAllowed() {
+        String className = "StringOnly";
+        Realm realm = Realm.getInstance(config);
+
+        realm.beginTransaction();
+        RealmObjectSchema objectSchema = realm.getSchema().get(className);
+
+        try {
+            objectSchema.addField("foo", String.class, FieldAttribute.PRIMARY_KEY);
+            fail();
+        } catch (UnsupportedOperationException ignored) {
```

Comment:
@cmelchior The place to throw exception is the same with `addPrimaryKey()`. So is it also feasible to throw `IllegalArgumentException` for `addPrimaryKey()`? it sounds a bit weird since `addPrimaryKey()` takes the field name as param, and it is not the reason of exception.